java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -386,14 +424,14 @@ private void applyOptionsHeaders(SdkHttpFullRequest.Builder builder, TransportOp
 
     private <ResponseT> ResponseT executeSync(
         SdkHttpClient syncHttpClient,
-        SdkHttpFullRequest httpRequest,
+        SdkHttpRequest httpRequest,
+        OpenSearchRequestBodyBuffer requestBody,
         Endpoint<?, ResponseT, ?> endpoint,
         TransportOptions options
     ) throws IOException {
-
         HttpExecuteRequest.Builder executeRequest = HttpExecuteRequest.builder().request(httpRequest);
-        if (httpRequest.contentStreamProvider().isPresent()) {
-            executeRequest.contentStreamProvider(httpRequest.contentStreamProvider().get());
+        if (requestBody != null) {
```

Comment:
Why we cannot reuse the `contentStreamProvider` (as before) but have to pass `requestBody` and create a new `ContentStreamProvider` ?